### ISSUE TITLE
Option "Build with Apache Ant" failing if path to build.xml file contains spaces

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#1]
**Description:**
If path to build.xml script contains spaces my build is failing. 

Path Example: d:\Moonshine Project\Moonshine-IDE\ide\MoonshineDESKTOPevolved\build\

**Environment:**
Windows 10

**Analysis:**
NativeProcessStartupInfo cannot properly parse provided path to build.xml file. I see following code which is responsible for wrong conversion:

    processArgs.push("set FLEX_HOME="+SDKstr+"&&"+ANTstr+" -file "+ UtilsCore.convertString(buildDir.fileBridge.nativePath) );

I did following fix which worked:

    processArgs.push("set FLEX_HOME="+SDKstr+"&&"+ANTstr+" -file \"" +  buildDir.fileBridge.nativePath + "\"");

Basically path now is surrounded by quotation marks and cmd is accepting it properly. I didn't make pull request, cause 

Unfortunately I'm not sure what repercussion may have removing UtilsCore.convertString(buildDir.fileBridge.nativePath) that is why I prefer raise defect.

**Comment by JustinProminic** [comment]
I think its safe to remove UtilsCore.convertString(..) in AntBuildPlugin usage. If adding the double-quotes in the manner it needs to fix the path spacing problem, then you please make a pull request. Please, test across the platforms before a commit (Windows, OSX).

**Comment by rat-moonshine** [comment]
I think its safe to remove UtilsCore.convertString(..) in AntBuildPlugin usage. If adding the double-quotes in the manner it needs to fix the path spacing problem, then you please make a pull request. Please, test across the platforms before a commit (Windows, OSX).

**Comment by JustinProminic** [comment]
I updated from Git. 

I tested with both type of projects which has space in it's path and not. But the Ant build failed for both of them. I created new ActionScript project with Moonshine and tried to build with Ant (Project > Build with Apache Ant).

Both reported similar errors in Moonshine console:

```
: Ant build Running
: Buildfile: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: checkOS:
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: init:
:     [mkdir] Created dir: C:\Users\Santanu\Documents\NewActionscriptXProject\build\DEPLOY
:     [mkdir] Created dir: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build-debug
: 
: modifyDescriptor:
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: createCertificate:
:      [echo] Self-signed .p12 certificate generation
:      [java] Error: Unable to access jarfile E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 \lib\adt.jar
: 
: 
: BUILD FAILED
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:72: The following error occurred while executing this line:
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:121: The following error occurred while executing this line:
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:86: Java returned: 1
: 
: Total time: 0 seconds
: 
: BUILD FAILED
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:72: The following error occurred while executing this line:
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:121: The following error occurred while executing this line:
: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml:86: Java returned: 1
: 
: Total time: 0 seconds
: FSCH exit code: 1
```

Please, check.

**Comment by rat-moonshine** [comment]
I updated from Git. 

I tested with both type of projects which has space in it's path and not. But the Ant build failed for both of them. I created new ActionScript project with Moonshine and tried to build with Ant (Project > Build with Apache Ant).

Both reported similar errors in Moonshine console:

```
: Ant build Running
: Buildfile: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: checkOS:
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: init:
:     [mkdir] Created dir: C:\Users\Devsena\Documents\NewActionscriptXProject\build\DEPLOY
:     [mkdir] Created dir: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build-debug
: 
: modifyDescriptor:
:      [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 
:   [taskdef] Could not load definitions from resource flexTasks.tasks. It could not be found.
: 
: createCertificate:
:      [echo] Self-signed .p12 certificate generation
:      [java] Error: Unable to access jarfile E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0 \lib\adt.jar
: 
: 
: BUILD FAILED
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:72: The following error occurred while executing this line:
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:121: The following error occurred while executing this line:
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:86: Java returned: 1
: 
: Total time: 0 seconds
: 
: BUILD FAILED
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:72: The following error occurred while executing this line:
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:121: The following error occurred while executing this line:
: C:\Users\Devsena\Documents\NewActionscriptXProject\build\build.xml:86: Java returned: 1
: 
: Total time: 0 seconds
: FSCH exit code: 1
```

Please, check.

**Comment by piotrzarzycki21** [comment]
I see in your log that changes itself working, cause build.xml has been successfully run. Couple of questions:

1) Can you run the same build from console ? Remember to setup in your system environment FLEX_HOME, same as I see in log.
2) Can you run the same build (no ANT) in Moonshine using that SDK ?

Please check and tell me whether the results are similar.

**Comment by rat-moonshine** [comment]
1. I have FLEX_HOME setup in environment variable in Windows - when running through command-line the Ant build using the same; Following are two different reports one using project path having space and another none. 

**Project path with space:**
```
E:\New folder\NewActionscriptProject\build>ant
Buildfile: E:\New folder\NewActionscriptProject\build\build.xml
     [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0

checkOS:
     [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0

init:
   [delete] Deleting directory E:\New folder\NewActionscriptProject\build\DEPLOY

   [delete] Deleting directory E:\New folder\NewActionscriptProject\build\build-
debug
    [mkdir] Created dir: E:\New folder\NewActionscriptProject\build\DEPLOY
    [mkdir] Created dir: E:\New folder\NewActionscriptProject\build\build-debug

modifyDescriptor:
     [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0

createCertificate:
     [echo] Self-signed .p12 certificate generation
     [copy] Copying 1 file to E:\New folder\NewActionscriptProject\build\build-d
ebug
     [copy] Copying 1 file to E:\New folder\NewActionscriptProject\build\build-d
ebug

compileSWF:
    [mxmlc] Loading configuration file E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22
.0\frameworks\air-config.xml
    [mxmlc] E:\New folder\NewActionscriptProject\build\build-debug\NewActionscri
ptProject.swf (794 bytes)

compileAIR:
     [echo] AIR build process
     [java] E:\New folder\NewActionscriptProject\build\build-debug\NewActionscri
ptProject-app.xml: error 102: Invalid namespace http://ns.adobe.com/air/applicat
ion/10.0

BUILD FAILED
E:\New folder\NewActionscriptProject\build\build.xml:72: The following error occ
urred while executing this line:
E:\New folder\NewActionscriptProject\build\build.xml:161: Java returned: 12

Total time: 10 seconds
```

**Project path without space:**
```
C:\Users\Santanu\Documents\NewActionscriptXProject\build>ant
Buildfile: C:\Users\Santanu\Documents\NewActionscriptXProject\build\build.xml
     [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0

checkOS:
     [echo] SDK : E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22.0

init:
   [delete] Deleting directory C:\Users\Santanu\Documents\NewActionscriptXProjec
t\build\DEPLOY
   [delete] Deleting directory C:\Users\Santanu\Documents\NewActionscriptXProjec
t\build\build-debug
    [mkdir] Created dir: C:\Users\Santanu\Documents\NewActionscriptXProject\buil
d\DEPLOY
    [mkdir] Created dir: C:\Users\Santanu\Documents\NewActionscriptXProject\buil
d\build-debug

modifyDescriptor:
     [copy] Copying 1 file to C:\Users\Santanu\Documents\NewActionscriptXProject
\build\build-debug
     [copy] Copying 1 file to C:\Users\Santanu\Documents\NewActionscriptXProject
\build\build-debug
     [copy] Copying 1 file to C:\Users\Santanu\Documents\NewActionscriptXProject
\build\build-debug

compileSWF:
    [mxmlc] Loading configuration file E:\DevareaLocal\ApacheFlexSDK\4.15_AIR.22
.0\frameworks\air-config.xml
    [mxmlc] C:\Users\Santanu\Documents\NewActionscriptXProject\build\build-debug
\NewActionscriptXProject.swf (797 bytes)

compileAIR:
     [echo] AIR build process

compileWindowsSignedPackage:
     [echo] Normal Signed Process Running
     [java] Signing options required to package from descriptor and fileset
     [java] usage:
     [java]   adt -checkstore SIGNING_OPTIONS
     [java]   adt -certificate -cn <name> ( -ou <org-unit> )? ( -o <org-name> )?
 ( -c <country> )? ( -validityPeriod <years> )? 2048-RSA <pfx-file> <password>
     [java]   adt -help
     [java]   adt -migrate SIGNING_OPTIONS ( <air-file-in> | <airn-file-in> ) <o
utput-file>
     [java]   adt -package SIGNING_OPTIONS ( -target air )? <output-package> ( <
app-desc> FILE_OPTIONS | <input-package> )
     [java]   adt -package SIGNING_OPTIONS -target airn <output-package> ( <app-
desc> FILE-AND-PATH-OPTIONS | <input-package> )
     [java]   adt -package -target ( apk | apk-debug | apk-emulator | apk-captiv
e-runtime ) ( CONNECT_OPTIONS? | LISTEN_OPTIONS? ) ( -airDownloadURL <url> )? (
ARCH_OPTIONS )? SIGNING_OPTIONS <output-package> ( <app-desc> PLATFORM-SDK-OPTIO
N? FILE-AND-PATH-OPTIONS | <input-package> PLATFORM-SDK-OPTION? )
     [java]   adt -package -target ( ipa-test | ipa-debug | ipa-app-store | ipa-
ad-hoc | ipa-test-interpreter | ipa-debug-interpreter | ipa-test-interpreter-sim
ulator | ipa-debug-interpreter-simulator ) ( CONNECT_OPTIONS? | LISTEN_OPTIONS?
) ( -sampler )? ANE_LINK_OPTIONS? AOT_MODE_OPTIONS? SIGNING_OPTIONS <output-pack
age> ( <app-desc> PLATFORM-SDK-OPTION? FILE-AND-PATH-OPTIONS | <input-package> P
LATFORM-SDK-OPTION? )
     [java]   adt -package SIGNING_OPTIONS? -target native SIGNING_OPTIONS? <out
put-package> ( <app-desc> FILE-AND-PATH-OPTIONS | <input-package> )
     [java]   adt -package SIGNING_OPTIONS? -migrate SIGNING_OPTIONS -target nat
ive SIGNING_OPTIONS? <output-package> <app-desc> FILE_OPTIONS PATH-OPTION
     [java]   adt -package SIGNING_OPTIONS? -target bundle SIGNING_OPTIONS? <out
put-package> ( <app-desc> FILE-AND-PATH-OPTIONS | <input-package> )
     [java]   adt -package SIGNING_OPTIONS? -target ane <output-package> <ext-de
sc> ANE_OPTIONS
....
....

```

2. I'm not sure what you did meant to run same build but no ANT in Moonshine using same SDK - did you meant to check by "Build & Run" option? If so then answer is, yes it works.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine,

Fix has been pushed for Windows. Project is building for me, but also I'm getting fail on ant target "compileWindowsSignedPackage". 

Let me know about the results.

**Comment by piotrzarzycki21** [comment]
Today I did tests on Mac using version 1.5.1 and it seems that part responsible for building apps by ant script working fine either with paths with spaces or not. 

Scripts for new Flex/AIR project built successfully swf, but failed with info: 
": compileDMGProject: :      [java] Native signing not supported on mac " - If you are going to check that and it is something which can we avoid - That would be separate issue.

**Comment by JoelProminic** [comment]
I confirmed that the Ant build works for AIR and Flex SWF projects with spaces in the path on Mac OS 10.10.3.

However, while testing this, I found that the Ant script for FlexJS, with or without spaces.  I will report this separately.
